src/ddmd/expression.d

```diff
@@ -8772,7 +8773,7 @@ extern (C++) final class DotIdExp : UnaExp
                     if (v.type.ty == Terror)
                         return new ErrorExp();
 
-                    if ((v.storage_class & STCmanifest) && v._init)
+                    if ((v.storage_class & STCmanifest) && v._init && !wantsym)
```

Comment:
To improve legibility of frontend code, it would help a lot if you use _much_ longer names for things. How about a method `bool mayReplaceSymbolWithInitializer()` ?